sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -63,24 +68,56 @@ object CosmosAccountConfig {
       val url = new URL(accountEndpointUri)
       val separatorIndex = url.getHost.indexOf('.')
       if (separatorIndex > 0) {
-        url.getHost.substring(0, separatorIndex)
+          url.getHost.substring(0, separatorIndex)
       } else {
-        url.getHost()
+          url.getHost()
       }
     },
     helpMessage = "Cosmos DB Account Name")
 
+  private[spark] val ApplicationName = CosmosConfigEntry[String](key = "spark.cosmos.applicationName",
+    mandatory = false,
+    parseFromStringFunction = applicationName => applicationName,
+    helpMessage = "Application name")
+
+  private[spark] val UseGatewayMode = CosmosConfigEntry[Boolean](key = "spark.cosmos.useGatewayMode",
+    mandatory = false,
+    parseFromStringFunction = useGatewayMode => useGatewayMode.toBoolean,
+    helpMessage = "Use gateway mode for the client operations")
+
   private[spark] def parseCosmosAccountConfig(cfg: Map[String, String]): CosmosAccountConfig = {
     val endpointOpt = CosmosConfigEntry.parse(cfg, CosmosAccountEndpointUri)
     val key = CosmosConfigEntry.parse(cfg, CosmosKey)
     val accountName = CosmosConfigEntry.parse(cfg, CosmosAccountName)
+    val applicationName = CosmosConfigEntry.parse(cfg, ApplicationName)
+    val useGatewayMode = CosmosConfigEntry.parse(cfg, UseGatewayMode)
 
     // parsing above already validated these assertions
     assert(endpointOpt.isDefined)
     assert(key.isDefined)
     assert(accountName.isDefined)
 
-    CosmosAccountConfig(endpointOpt.get, key.get, accountName.get)
+    CosmosAccountConfig(
+      endpointOpt.get,
+      key.get,
+      accountName.get,
+      applicationName,
+      useGatewayMode.getOrElse(false))
+  }
+}
+
+private[spark] case class CosmosReadConfig(forceEventualConsistency: Boolean)
+
+private[spark] object CosmosReadConfig {
+  val ForceEventualConsistency = CosmosConfigEntry[Boolean](key = "spark.cosmos.read.forceEventualConsistency",
```

Comment:
CosmosConfigEntry has infra for supporting default value, could you please use that for passing the default.

default for forceEventualConsistency should be true.

see sample please:
https://github.com/moderakh/azure-sdk-for-java/blob/eb13c04777dd8fb183f30e4b217e6c3bba573dcc/sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala#L102

Reply:
Done